### ISSUE TITLE
Wrong load order for custom locale files

### ISSUE THREAD

**Issue by usabi** [consul/consul#1231]
Hi!

We found that the configuration of locale load paths in config/application.rb causes custom locale definitions to be masked by original ones.

It seems that the configuration is loading all files in config/locales folder and subfolders, but the priority is somehow wrong. Inspecting I18n.load_paths we found that the 'config/locales' files where duplicated but in the right order, that is, in the wrong case we see this (for one of the files involved)
```
(byebug) I18n.load_path.grep /activerecord\.es/
[".../consul/config/locales/activerecord.es.yml", ".../consul/config/locales/activerecord.es.yml", ".../consul/config/locales/custom/activerecord.es.yml"]
```
but in the right case, we see this instead
```
(byebug) I18n.load_path.grep /activerecord\.es/
[".../consul/config/locales/activerecord.es.yml", ".../consul/config/locales/custom/activerecord.es.yml"]
```

The problem doesn't seem to be the order in the array, but the order in which the files are processed to form the locales hash.

Besides, the paths related to custom models, controllers and views are included separately, so we think is better to do the same with locales.

Including only this line below the others the problem is fixed.

``` 
config.i18n.load_path += Dir[Rails.root.join('config', 'locales', 'custom', '*.{rb,yml}')]
```

Included in the pull request, there is a couple of tests, but we are not fully confident that they reflect the real situation. Evaluating I18n.load_paths gives us the duplicity but not in the same order. Maybe someone knows what is happening exactly.

**Comment by usabi** [comment]
Keep thinking about this... maybe the problem is that rails locales are not prepared to have duplicated keys unless they belong to different language, and the order in which the system processes the files to form the translations is not predefined.

**Comment by voodoorai2000** [comment]
Maybe... in any case if it's working and tested we are happy to close this issue for now. 
Thanks for the PR! 

Closed via #1232
